packages/back-end/src/api/saved-groups/deleteSavedGroup.ts

```diff
@@ -0,0 +1,28 @@
+import { DeleteSavedGroupResponse } from "../../../types/openapi";
+import {
+  deleteSavedGroupById,
+  getSavedGroupById,
+} from "../../models/SavedGroupModel";
+import { createApiRequestHandler } from "../../util/handler";
+import { deleteSavedGroupValidator } from "../../validators/openapi";
+
+export const deleteSavedGroup = createApiRequestHandler(
+  deleteSavedGroupValidator
+)(
+  async (req): Promise<DeleteSavedGroupResponse> => {
+    const savedGroup = await getSavedGroupById(
+      req.params.id,
+      req.organization.id
+    );
+
+    if (!savedGroup) {
```

Comment:
@jdorn I added the delete endpoint and modeled it after Stripe where the return is the id of the item that was deleted. 

I was curious on your opinion of this implementation - currently, across our code base we don't seem to do any sort of validation on `deleteOne` calls, since I think we typically call `mutate`. 

But in this scenario, it felt necessary to check if the `savedGroup` exists before we call delete, so we don't just tell the user it was deleted when it's already been deleted.

If we don't like this implementation, I can always update the `deleteSavedGroupById` to return the `deletedCount` and if `deletedCount === 0` throw a really generic error, otherwise I can return the id.

Reply:
I'm fine with checking if it exists first and throwing an error if not.  Returning the deleted id doesn't seem very useful, so not sure why Stripe is doing it. You need to know the id already in order to call the DELETE endpoint.  I guess it gives the user a little more confidence that it was actually deleted.  This is how Stripe structures the return value, which is slightly different.  Instead of `deletedId`, they have separate `id` and `deleted` fields that are returned.

```json
{
  "id": "...",
  "deleted": true
}
```

I don't have any strong preferences here, I think both approaches accomplish the same thing.